Add smuggler stashes

IamVelcroboy opened the pull request:
## About the PR
- Adds smuggler stashes in the form of satchels that ~~can me mapped in with an rng spawner~~ will spawn randomly from a roundstart event. 
- Satchels will contain rndm contents from a large pool of items.
- Satchels are hidden under tiles and can be located with a t-ray device.
- There's a very small chance that instead of a satchel, there will be a briefcase full of money.
- This also changes all bags and briefcases so they can now be hidden under tiles by anchoring them. 
- Update: Pools have been split up. This has several advantages. Results are a little less crazy. Rare items will be even more rare. Instances where bags might spawn more item weight than the bag can hold no longer occurs. Provides an easy method for downstreams to add their own loot pools. 
- Update: Adds a component to prevent stacking bags in the same tile. Taken with permission from [Stashes](https://github.com/new-frontiers-14/frontier-station-14/pull/2347)

## Why / Balance
Found this feature on TG wiki and thought it sounded super fun. Balance can be found by how they are mapped. Mapping <20 spawners on a station means not every round will have a hidden stash. Mapping 20 spawners means there should be 1 every round(on average of course.) Lots of time/effort went into balancing the pools but they are random so they could occasionally end up with some wild contents. This should be very rare though which, imo, makes it extra cool. 

I also think the ability to stash regular bags will add a lot of fun, as well as new tactical choices for antags. 

## Technical details
n/a

## Media

https://github.com/space-wizards/space-station-14/assets/107660393/098a847d-55e2-4d93-a141-6147c585fa2a

10 randomly generated satchels: (Updated 11/04/24)
![image](https://github.com/user-attachments/assets/0c56045a-0102-4dd7-89a5-0dad953e6f10)
![image](https://github.com/user-attachments/assets/c08aabeb-da0b-48c5-bd82-d4e7aba9d7df)
![image](https://github.com/user-attachments/assets/62d048b6-efe4-49cb-a2d8-9f0535a21c67)
![image](https://github.com/user-attachments/assets/12b0064f-9142-48f7-84ce-8a8beb69b9f0)
![image](https://github.com/user-attachments/assets/9bbe4147-d830-4d8d-b931-65f8a0f5f7ba)
![image](https://github.com/user-attachments/assets/4cfcf4d1-1d5d-4d98-baa4-df128a52cc36)
![image](https://github.com/user-attachments/assets/2827dcf8-101d-4d8f-b68e-c929e616cf98)
![image](https://github.com/user-attachments/assets/658d04a3-70dd-421f-99e9-fa5260b6633b)
![image](https://github.com/user-attachments/assets/2b253b41-ef16-4d19-a60b-98a844a4f0fa)
![image](https://github.com/user-attachments/assets/1168984e-da7b-4711-b410-db2758671dc6)


- [x] I have added screenshots/videos to this PR showcasing its changes in-game, **or** this PR does not require an in-game showcase

## Breaking changes
n/a

**Changelog**

:cl: Velcroboy, ErhardSteinhauer, whatston3, MilonPL, Beck, ArtisticRoomba, ScarKy0
- add: Smuggler stashes! Will get spawned as a round start event and can also be found in a hacked ClothesMate. Will have loot from a large random pool.
- add: Smuggler stashes can now be bought from the syndicate uplink for 2TC.

Equivocateur commented:
Cue "remove cat ears"

ChilbroBaggins commented:
Oi, didn't see that coming. Let me see what I can workaround...

deltanedas commented:
cue "remove blood-red encryption key"

ChilbroBaggins commented:
> cue "remove blood-red encryption key"

I might have to tweak the probs on the keys, they shouldn't have spawned that many to begin with. Either way, Syndi key is(will be) very rare. And what an interesting round if security gets their hands on it, no...?

Equivocateur commented:
Were not real maintainers were just predicting the eventual demand for culling gamer loot from the repo coppers.

> cue "remove blood-red encryption key"

There's an entire emag in there fyi

ChilbroBaggins commented:
> Were not real maintainers were just predicting the eventual demand for culling gamer loot from the repo coppers.
> 
> > cue "remove blood-red encryption key"
> 
> There's an entire emag in there fyi

Yeah, I get it. I expect some level of scrutiny here. I'm fully aware of the way many of these items can change the flow of the round. I do however believe it will be interesting given it can be properly balanced. It certainly should not be every round someone finds a high-impact antag item. But occasionally it should be fun. And if mapped well enough, they should not be overly gameable. They should be rare and random in both contents and location from round to round.

ChilbroBaggins commented:
OK, I'm gonna need to dig into this collision thing a bit. But I have hungry cats. So...gonna be a lil bit.

NancokPS2 commented:
Would a powerful chem like nocturine or omnizine with a very low chance be a problem?

deltanedas commented:
also it should not have a unique name because anyone with a t-ray will instantly meta it

ChilbroBaggins commented:
> also it should not have a unique name because anyone with a t-ray will instantly meta it

Is a good point, however, it does have a different sprite. So, could still be noticed anyhow.

Emisse commented:
remove cat ears

ChilbroBaggins commented:
> remove cat ears

Haha, really? I genuinely can't tell if that's a joke or not. 😂Keep in mind I can always increase the rarity if need be.

metalgearsloth commented:
remove cat ears

ChilbroBaggins commented:
wow, it's that real huh?

NancokPS2 commented:
Cat ears are considered the ultimate rarity in lore so it would be unrealistic.
-Source: I made it the fuck up.

ChilbroBaggins commented:
I mean, I'll take it out. Not gonna fuss about it. But just out of curiosity, if it was made configured to show up something like 1 out of 100 rounds, or 1000, or whatever....is it still not an option? 

Like I said, it's really not a big deal to me but I _am_ curious.

deltanedas commented:
dont care you are valid

ChilbroBaggins commented:
I updated the loot pools to be a little less crazy. This way will also provide a way for down-streams to add their own loot pools easily.

ChilbroBaggins commented:
Ok, tests are passing and I've got this in a pretty good place balance-wise. (Considering decreasing the rareChance of the briefcase from .001->.01. Will wait for opinion on that though.)

Otherwise, I think I'm done fussing with it until a review is made. Let me know if you want me to run more satchel gens to see a better picture of how they look.

EmoGarbage404 commented on Resources/Prototypes/Entities/Clothing/Back/backpacks.yml:
```diff
@@ -9,6 +9,13 @@
     size: 9999
+  - type: Appearance
+  - type: SubFloorHide
+  - type: Anchorable
+  - type: CollideOnAnchor
+    enable: false
```
backpacks shouldn't be anchorable

metalgearsloth replied:
None of these extra comps should be here.

ChilbroBaggins replied:
Removing that would completely negate half the feature of this PR, Making bags stashable. Note: That you can't simply anchor to _any_ tile. It can only be anchored to plating, where you can then place a tile over it. Effectively stashing it.

metalgearsloth reviewed (changes requested):
Don't add dummy components to prototypes please, use parent composition if you really need.

ChilbroBaggins commented:
> Don't add dummy components to prototypes please, use parent composition if you really need.

Where did I do that?

metalgearsloth commented:
> > Don't add dummy components to prototypes please, use parent composition if you really need.
> 
> Where did I do that?

![image](https://github.com/space-wizards/space-station-14/assets/31366439/79d90246-9038-4ac0-a83d-10563e4f4185)

Bags shouldn't just globally be anchorable.

ChilbroBaggins commented:
> Bags shouldn't just globally be anchorable.

That's what makes them "stashable." Did you watch the video to see how this works?
![image](https://github.com/space-wizards/space-station-14/assets/107660393/bd6fcbd3-f36d-40d2-9a5a-837bfa1b9379)

0x6273 commented:
Why not make it so putting a tile on a space with an item puts the item under the tile instead of on top? Using anchoring for this seems really weird.

ChilbroBaggins commented:
> Why not make it so putting a tile on a space with an item puts the item under the tile instead of on top? Using anchoring for this seems really weird.

Honestly, while testing this, it felt quite natural. It feels like you're actually stuffing the bag down into the tiles. I wouldn't want to remove the do-after from the action. 

I'm probably not going to do it the way you suggest. It would be significantly more difficult and open a can of worms concerning item sizes and what can/cannot have a tile placed on it.

metalgearsloth commented:
> That's what makes them "stashable." Did you watch the video to see how this works?

I watched the video and anchoring shouldn't be used for this.

ChilbroBaggins commented:
> I watched the video and anchoring shouldn't be used for this.

What would you suggest then? I don't really see another way to make it work w/out introducing some new mechanic. 

Anchoring works quite well for this. I'm not sure why it would be an issue?

metalgearsloth commented:
> > I watched the video and anchoring shouldn't be used for this.
> 
> What would you suggest then? I don't really see another way to make it work w/out introducing some new mechanic.
> 
> Anchoring works quite well for this. I'm not sure why it would be an issue?

- not intuitive
- A lot of code doing lookups isn't going to handle this specific use-case
- I don't think this handles construction.
- Component bloat for a very frequent item
- makes bag interactions even more inconsistent
- I think it's the only pickupable anchorable item in the game which is going to cause bugs for something rarely used
- Need to add checks around other anchorable code that may be unique on its own tile
- Layering issues

is the short-list.

CaptainSqrBeard commented:
> * I think it's the only pickupable anchorable item in the game which is going to cause bugs for something rarely used
we have pipes that anchorable and could be picked up and they didn't cause any bugs

ChilbroBaggins commented:
It easily becomes intuitive as it's learned. Like I mentioned to the other user, it doesn't feel unnatural at all, imo.

Not sure what construction has to do with it?

I don't see it as comp-bloat if it's serving a purpose. The purpose here is a fun new way to hide things.

Only inconsistency really is that, when a bag is anchored, you only need to click on it to open UI. (Which is actually kinda cool.) Vs. context menu opening.

Yes, as CaptainSqrBeard mentioned, pipes use this same functionality w/out issue. 

Checks could be added, sure. Are there similar checks for pipes and other undertile entities? (I'm not at a PC right now to really look into that.)

Layering issues feels like a bit of a stretch. But it could be addressed if it becomes something noticeable. I didn't notice anything while testing. 

I don't know, I feel like if you gave it a chance, you'd like what this PR does, and players would too. But if you have any suggestions how I could improve it or avoid any potential issues, please let me know. I'm not really sure I know of another way, short of creating a whole new function just for this. 

I'm away from my PC until next week. So, I won't be able to make any changes until then anyway. So, there's plenty of time to think on it.

metalgearsloth commented:
> > * I think it's the only pickupable anchorable item in the game which is going to cause bugs for something rarely used
> 
> we have pipes that anchorable and could be picked up and they didn't cause any bugs

Pipes are very buggy wym.

CaptainSqrBeard commented:
> > > * I think it's the only pickupable anchorable item in the game which is going to cause bugs for something rarely used
> > 
> > 
> > we have pipes that anchorable and could be picked up and they didn't cause any bugs
> 
> Pipes are very buggy wym.

I haven't seen any bugs related to them being anchorable and also item.

NancokPS2 commented:
Probably the fact that they can be infinitely stacked? Even tho a fix is on the way.    

It may not even matter in this case anyway, idk.

GeorgeGimme commented:
i like the anchoring its cool

ChilbroBaggins commented:
Just fyi, I do plan to come back to this, I have an idea to make this not rely on anchoring. (Unless you change your mind about that😁)

But, it will be a couple weeks until I can really get into it. I'm a little busy with some other projects atm.

JoeHammad1844 commented:
still working on this?

moonheart08 commented:
I'm perfectly okay with you using anchoring personally.

IamVelcroboy commented:
> I'm perfectly okay with you using anchoring personally.

That would certainly make things easier. IRL had been busier than I expected. So, working on some SubfloorHideOnTilePlacement comp is going to take longer than just fixing the merge conflict....

deltanedas commented:
dont pipes already do that...

JoeHammad1844 commented:
troll the captain by anchoring their backpack in the bar

IamVelcroboy commented:
Fixed merge conflict. Test fail seems to be the same false-positive showing up on my other PRs. Ignoring unless it's relevant.

Also, not completely sure if the direction of this PR is still up for discussion...? If there's an argument for keeping it as-is, that'd be preferable imo. But if the absolute answer is "no" I can probably start working on something this weekend.

JoeHammad1844 commented:
smugglers satchels full of bananium for epic trolling

TheShuEd commented:
how's the development going?

IamVelcroboy commented:
> how's the development going?

I am still coming back to this. Although, it would be nice to get some clarification. I asked in one of the comments above and also in Discord [here](https://discord.com/channels/310555209753690112/1112470595284783247/1182015382421258321)

Peptide90 commented:
> > how's the development going?
> 
> I am still coming back to this. Although, it would be nice to get some clarification. I asked in one of the comments above and also in Discord [here](https://discord.com/channels/310555209753690112/1112470595284783247/1182015382421258321)

Looks good to me but I ain't a maintainer (surprised pikachu). Requested a review on discord?

IamVelcroboy commented:
Alright, I spent some time with this addressing the comments and bringing it up to speed with gridinv. 

I still don't have a solid direction to move on this but the more I test it and the more I consider making changes, the more I strongly believe this current method is the way to go. It just works. It's intuitive and it's fun. 

Adding in a whole new system with new comps and updating entities to support it just seems completely unnecessary and semi-redundant.  But, I'll still move in that direction if it's absolutely necessary.

IamVelcroboy commented:
Alright, I spent some time with this addressing the comments and bringing it up to speed with gridinv.

I still don't have a solid direction to move on this but the more I test it and the more I consider making changes, the more I strongly believe this current method is the way to go. It just works. It's intuitive and it's fun.

Adding in a whole new system with new comps and updating entities to support it just seems completely unnecessary and semi-redundant. But, I'll still move in that direction if it's absolutely necessary.

beck-thompson commented:
Are you still working on this? It would be super cool!

IamVelcroboy commented:
> Are you still working on this? It would be super cool!

Yep, just waiting for answers to my questions above ☝

Emisse commented:
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

IamVelcroboy commented:
Can this be reopened please?

TheShuEd commented:
Sure

IProduceWidgets commented:
I like the idea. I wonder if ~20 spawn locations is enough to stop people from memorizing them though. That certainly happens with other games.

IamVelcroboy commented:
> I like the idea. I wonder if ~20 spawn locations is enough to stop people from memorizing them though. That certainly happens with other games.

I think mappers could get really creative with hiding them. But also, I see no issue making that number higher. I'm definitely open to suggestions if that seems too low. It's also something that could be adjusted down the line at any time with a simple PR.

SlamBamActionman commented:
I like this idea on paper but I am trying to see how this will play out ingame. Tiders will undoubtedly learn all the spawn spots and incorporate them into the normal start-of-round pathing they take when looting maints lockers. I guess the difference is they will be walking around with an active t-ray in their pocket?

Having them also mapped in departments might make them a bit less powergamed though, since those who play jobs tend to focus on that rather than getting a t-ray just to check a single 5%ish spawn.

SlamBamActionman commented:
Thinking about this more, the "hide bag under tile" feature will be a very strong traitor/thief tool. It will be virtually impossible to locate a stash hidden under a tile under a wall in some dark maints corner.

(Unless t-rays go through walls, at which point disregard this comment)

IProduceWidgets commented:
I've seem some games address that with a mercy mechanic. Something like a little sparkle if you've been (still?) in an area for a while.

SlamBamActionman commented:
Maybe it is easier to prevent walls being built/stuff being anchored on a tile that has a bag under it?
"The flooring underneath appears unstable"
That way t-rays will always be able to show it and you don't have to deconstruct entire walls to check a specific tile.

IamVelcroboy commented:
I think to some degree, power gamers scouring every known spawn location is mitigated by the bag randomness itself.

The 1(very occasionally 2) bags in any given round will still be randomized themselves. So, even if you're the first to find the bag every round. It could be many rounds before you get one filled with something powerful.

Lots of times you'll get burgers and cigarettes and maybe a cool circuit board. But again, I think it's going to come down to some creative mapping and maybe decreasing the odds of a given spawn even further, as suggested above.

TheShuEd commented:
I suggest not making specific spawn points, but randomly spawning them in atmos pipes locations

IProduceWidgets commented:
> I suggest not making specific spawn points, but randomly spawning them in atmos pipes locations

Would have to also check for walls, windows, and that the there's a tile. But yeah that could work.

jorgun commented:
making and event and using randomspawnrule(same as ratking spawn location) will just put it on a random tile.
gets aroud the known location powergaming problem but it can spawn inside maint and onto tiles missing floorbords

IamVelcroboy commented:
Hi, I see the suggestions here. Been pretty busy but I'll try to look into them next week.

IamVelcroboy commented:
OK, I switched this to run a roundstart event that spawns a satchel somewhere on the station randomly.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+        prob: 0.05
+        maxAmount: 5
+        orGroup: Chems
+      - id: VestineChemistryVial
+        prob: 0.05
+        orGroup: Chems
+      - id: OmnizineChemistryBottle
```
The only purpose for these is antag stuff, I don't think its a good idea to place them around the station where passengers will find them.

IProduceWidgets replied:
Oops I didn't mean to include the omni there, Omni is probably fine.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: CognizineChemistryBottle
+        prob: 0.05
+        orGroup: Chems
+      - id: ToxinChemistryBottle
+        prob: 0.05
+        orGroup: Chems
+      - id: NocturineChemistryBottle
+        prob: 0.05
+        orGroup: Chems
```
"

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+        prob: 0.02
+        maxAmount: 2
+        orGroup: GizmosPayload
+      - id: ExplosivePayload
+        prob: 0.02
+        maxAmount: 2
+        orGroup: GizmosPayload
```
"
Maybe would be okay as an IED or something

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: ChemDispenserMachineCircuitboard
+        prob: 0.05
+        orGroup: Circuitboards
+      - id: SyndicateMicrowaveMachineCircuitboard
+        prob: 0.05
+        orGroup: Circuitboards
```
I'm pretty sure this is just a less useful microwave than the normal one. It might be slightly faster? I know it cant provide access.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: ToyFigurineRatKing
+        prob: 0.01
+        orGroup: PresentsOrToys
+      - id: PresentRandomUnsafe
+        prob: 0.0015
+        orGroup: PresentsOrToys
```
This has the potential to give admin items and other things players probably shouldn't have access to regularly.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: EncryptionKeyService
+        prob: 0.02
+        orGroup: Keys
+      - id: EncryptionKeyRobo
+        prob: 0.02
+        orGroup: Keys
```
hmmm. This one is questionable.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: EncryptionKeyEngineering
+        prob: 0.02
+        orGroup: Keys
+      - id: EncryptionKeySyndie
+        prob: 0.02
+        orGroup: Keys
```
"

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: EncryptionKeyCommand
+        prob: 0.02
+        orGroup: Keys
+      - id: EncryptionKeyStationMaster
+        prob: 0.001
+        orGroup: Keys
```
I think this ones kind of funny, but also seems like it could be a bad idea since these will mostly be found by non-antags.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+        prob: 0.05
+        maxAmount: 10
+        orGroup: Syndicate
+      - id: Emag
+        prob: 0.05
+        orGroup: Syndicate
```
While this does have some uses that aren't terrible, I also don't know if its a good idea to give to non-antags.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: NukeDiskFake
+        prob: 0.05
+        orGroup: Syndicate
+      - id: RadioJammer
+        prob: 0.05
+        orGroup: Syndicate
```
There's not use for this as a non-antag that isn't just being annoying

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: RadioJammer
+        prob: 0.05
+        orGroup: Syndicate
+      - id: ClothingMaskGasVoiceChameleon
+        prob: 0.05
+        orGroup: Syndicate
```
"

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: ClothingHandsGlovesBoxingRigged
+        prob: 0.05
+        orGroup: Syndicate
+      - id: ThievingGloves
+        prob: 0.05
+        orGroup: Syndicate
```
"

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: HandheldGPSBasic
+        prob: 0.10
+        orGroup: Gear
+      - id: WelderIndustrialAdvanced
+        prob: 0.10
+        orGroup: Gear
```
should probably be in the tools group

IProduceWidgets replied:
oh nvm different case.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: HandTeleporter
+        prob: 0.02
+        orGroup: Gadgets
+      - id: DeviceQuantumSpinInverter
+        prob: 0.02
+        orGroup: Gadgets
```
This doesn't have a use unless you have two.

IProduceWidgets commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,1009 @@
+      - id: DeviceQuantumSpinInverter
+        prob: 0.02
+        orGroup: Gadgets
+      - id: HoloClownInjector
+        prob: 0.02
+        orGroup: Gadgets
+      - id: MagicalLamp
+        prob: 0.01
+        orGroup: Gadgets
```
No use for non-antags. There is an arabian lamp you could use :godo:

IamVelcroboy commented:
@IProduceWidgets Just wanted to discuss a few of your comments about antag items in the pool. 

A few points I see:
1. I kinda see these as something antags would seek out (of course tiders will also try to beat them to it) as it would be like little potential free gifts for them. So, having some good antag items incentivizes that.
 
2. Having some antag items appear on station outside of normal means also adds some interesting elements of doubt when trying to determine a players antag status. (Did they just find this item? Or are they really an infiltrator?)
     2b. That said, I understand there may be a few items that go too far in the wrong hands.

3. The rarity of the bags + the wide variability of the items inside + the lower chances for antag items vs standard items = the majority of rounds will not see these items. It will be very rare for them to occur. You may still be right that they shouldn't be there at all but just keep that in mind. 
     3b. I did lots and lots of testing/tweaking, spawning 20 bags at a time, over and over, checking each one to make sure items weren't too common. 
     
4. Non-antags might find these, true, but that includes command, sec, and players that would turn these over to security. Which is why I think it's interesting to have something like the syndie key, etc.
 
5. Lore-wise: a smuggler would be an antagonist and a stashed bag of theirs would likely have some contraband. 

Just my thoughts, or what I was thinking while adding them. But I'll definitely make the changes if it's still deemed necessary. Let me know what you think.

beck-thompson commented:
I agree with the antag item stuff. Right now if sec catches someone with illegal items they can basically know 100% they are an antag! Giving some more items out would help mitigate this (Especially ones that aren't super dangerous).

bruhmogus commented:
is this still being worked on?

IamVelcroboy commented:
> is this still being worked on?

Yes, I started a new job and it's been a bit hectic. I'll fix conflicts/reviews soon

IamVelcroboy commented:
Before comment/close, I am planning to work on this this weekend.

LevitatingTree commented:
hope this'll be added eventually, really promising

ScarKy0 commented:
RP changelogs are a no iirc

everydayeldergod commented:
Can we get a second look at this? I think it's fine to have the stash be globally anchorable - given that that's the point of the item, after all. I think there is broad public support and no real reason to delay any longer, especially since Goob has already implemented the Satchel and it works well there.

Hope to see this in the game soon!

IamVelcroboy commented:
Yeah, I'll fix conflict this weekend. If there's any additional feedback I can address it then too

IamVelcroboy commented:
Updated with a component to prevent stacking bags in the same tile. 
![image](https://github.com/user-attachments/assets/6008d1c8-5f14-4a32-bcd2-bdb23c0e028c)

Pumkin69 commented:
holy shit this is still being worked on?

beck-thompson commented on Content.Server/Storage/EntitySystems/AnchorableStorageSystem.cs:
```diff
@@ -0,0 +1,115 @@
+        return CheckOverlap((uid, comp, Transform(uid)));
+    }
+
+    public bool CheckOverlap(Entity<AnchorableStorageComponent, TransformComponent> ent)
```
Same here

beck-thompson commented on Content.Server/Storage/EntitySystems/AnchorableStorageSystem.cs:
```diff
@@ -0,0 +1,115 @@
+            return;
+
+        // Nothing around? We can anchor without issue.
+        if (!CheckOverlap((ent, ent.Comp, Transform(ent))))
```
It will still fail to anchor when there is a tile in the location (Makes sense!) but the comment confused me. Maybe changed it to like "No other stashes or anchored items?"

beck-thompson commented on Content.Server/Storage/EntitySystems/AnchorableStorageSystem.cs:
```diff
@@ -0,0 +1,115 @@
+                continue;
+
+            // Is another storage entity is already anchored here?
+            if (HasComp<AnchorableStorageComponent>(otherEnt))
```
Also probably make this a whitelist! I could imagine adding other items that can be anchored under tiles and you might not want to mix some.

beck-thompson commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,670 @@
+  #number after each group is potential max inventory capacity. If combined PMIC is too high, it could result in an over-filled bag/error.
+
+    #Group 1
+    - !type:AllSelector
```
Could you use nested selectors here so it isn't so big? You could split up the tables a lot easier and it would be much better for reading and editing.

beck-thompson commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,670 @@
+  table: !type:GroupSelector
+    children:
+    - id: ClothingBackpackSatchelSmugglerFilled
+      weight: 100
```
Maybe reduce this weight by a little so the others are slightly more common to appear?

beck-thompson reviewed (changes requested):
Looks good! The only other thing is you should move it all to shared. I don't think there should be any issues with it :thinking:

Also the `SubFloorHideSystem` should be moved to shared to prevent more miss-predictions (I think it shouldn't be too hard!) but that's out of scope for this pr!

ScarKy0 commented:
So after testing around there is one minor issue I've found with this
If a backpack/satchel/etc is worn you can no longer click insert wrenches into it as it starts an anchor attempt, and this probably shouldnt happen unless the bag can actively be anchored in a position

beck-thompson reviewed (changes requested):
PR is conceptually approved! There are only two requested changes:
1.) Remove all the syndicate contraband (E.g the chameleon projector, thieving gloves etc...). Maintainers don't want graytiders to be all running to maints to find these as soon as possible for gamer loot! Anything super powerful should be removed.
2.) Only allow the smuggler stash item (Or similar ones like the clown and briefcase) to be anchorable! E.g someones round start backpack shouldn't have that ability.

Let me know if you have any questions. Because of how old the PR is, I'll try to review it as fast as possible so it can finally get merged :)

SpeltIncorrectyl commented:
> PR is conceptually approved! There are only two requested changes: 1.) Remove all the syndicate contraband (E.g the chameleon projector, thieving gloves etc...). Maintainers don't want graytiders to be all running to maints to find these as soon as possible for gamer loot! Anything super powerful should be removed. 2.) Only allow the smuggler stash item (Or similar ones like the clown and briefcase) to be anchorable! E.g someones round start backpack shouldn't have that ability.
> 
> Let me know if you have any questions. Because of how old the PR is, I'll try to review it as fast as possible so it can finally get merged :)

I strongly agree that syndicate contra should be removed. Otherwise the most obnoxious person you know will spent the entire shift crowbaring the entire station to find gamer loot.

walksanatora commented:
Not a Maintainer but on Starlight as Captain had a round where people went around removing all the floor tiles. all I did was tell sec "it is minor vandalism" and then let sec start perma people for repeated offenses. and if you go ripping up all the floor tiles you WILL get called to sec and ordered to stop. making it a better idea to try and locate a T-ray before going stash hunting so you are less suspiscous.

ArtisticRoomba commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,108 @@
+  components:
+  - type: Sprite
+    layers:
+      - state: red
+      - sprite: Clothing/Back/Satchels/smuggler.rsi
+        state: icon
```
Is this right?

ArtisticRoomba commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,108 @@
+          - id: RevolverCapGunFake
+            amount: !type:RangeNumberSelector
+              range: 4, 8
+        - id: SpeedLoaderCap
+          amount: !type:RangeNumberSelector
+            range: 4, 8
```
Rearrange the SpeedLoaderCap to be at the top above the GroupSelector, right now it looks like the spacing is wrong but it's technically correct right now because of intent

ArtisticRoomba reviewed (changes requested):
Nitpicks, I'm probably going to just fix them all on my local and send you a patch file like we previously discussed.

beck-thompson commented:
For anyone reviewing this:
I did leave a minor amount of syndicate contraband. Nothing dangerous but I think its just:
Chameleon projector
1-5 TC
Radio jammer
Encryption keys (of any department at random)

I personally think minor stuff like that is fine (Just not stuff like holoclown implants, assault rifles etc...)

MilonPL commented on Content.Shared/Construction/Components/BlockAnchorOnComponent.cs:
```diff
@@ -0,0 +1,23 @@
+/// <summary>
+///     Will not allow anchoring if there is an anchored item in the same tile that fails the whitelist checks.
+/// </summary>
+[RegisterComponent, NetworkedComponent]
```
Should have an Access attribute for the relevant system.

MilonPL commented on Content.Shared/Storage/Components/AnchoredStorageFilterComponent.cs:
```diff
@@ -0,0 +1,27 @@
+/// <example>
+///     If you have a smuggler stash that has a player inside of it, you want to eject the player before its anchored so they don't get stuck
+/// </example>
+[RegisterComponent, NetworkedComponent]
```
Access.

MilonPL commented on Content.Shared/Storage/EntitySystems/AnchoredStorageFilterSystem.cs:
```diff
@@ -0,0 +1,44 @@
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+
+    public override void Initialize()
```
Missing the base call here.

MilonPL commented on Content.Shared/Construction/EntitySystems/BlockAnchorOnSystem.cs:
```diff
@@ -0,0 +1,69 @@
+    [Dependency] private readonly SharedTransformSystem _xform = default!;
+    [Dependency] private readonly EntityWhitelistSystem _whitelist = default!;
+
+    public override void Initialize()
```
Also missing the base call.

MilonPL commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,107 @@
+- type: entity
+  id: SubfloorAnchorStorageBase
+  abstract: true
```
Abstract should always be the first thing after `type`

MilonPL commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,107 @@
+- type: entity
+  parent: SubfloorAnchorStorageBase
+  id: AnchoredSubfloorAnchorStorageBase
+  abstract: true
```
Above.

MilonPL commented on Resources/Prototypes/Entities/Clothing/Back/smuggler.yml:
```diff
@@ -0,0 +1,107 @@
+- type: entity
+  id: RandomSatchel
+  name: random smuggler's satchel spawner
+  parent: MarkerBase
```
Parent > id > name

MilonPL commented on Resources/Prototypes/GameRules/events.yml:
```diff
@@ -640,6 +640,19 @@
 
+- type: entity
+  id: SmugglerStashVariationPass
+  parent: BaseGameRule
```
Move parent above.

MilonPL commented on Resources/Prototypes/GameRules/events.yml:
```diff
@@ -640,6 +640,19 @@
+  id: SmugglerStashVariationPass
+  parent: BaseGameRule
+  components:
+    - type: StationEvent
```
Wrong indentation.

MilonPL commented on Resources/Textures/Clothing/Back/Satchels/smuggler.rsi/meta.json:
```diff
@@ -0,0 +1,26 @@
+{
```
Use 4 space indentation for json files, not 2.

beck-thompson replied:
pretty sure its 4! I could be losing it though

ArtisticRoomba replied:
This file is 4 spaced goober

ArtisticRoomba replied:
Oh wait you messed up the spacing. Some parts are inconsistent.

Just let rider do it for you godo

MilonPL reviewed (approved):
Looks good to me.

ScarKy0 reviewed (changes requested):
Looks good. Few issues around the names.
Since It's impossible for this to be a stealth item (huge "THIS IS UNANCHORED" text in examine) it should probably be all reflected in the name and contraband status.
Other maints asked me to just address these myself so I will likely do it later.

ScarKy0 commented:
Also contraband inventory isnt emagged inventory, its what you get when you hack the MNGR wire, but i think itll be a bit more interesting like this

ScarKy0 reviewed (approved):
Cool stuff :3

MilonPL commented:
shortest wizden review

ScarKy0 commented:
freedom

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Back/smuggler_tables.yml:
```diff
@@ -0,0 +1,673 @@
+    - id: ToyFigurineSpaceDragon
+    - id: ToyFigurineQueen
+    - id: ToyFigurineRatKing
+    - id: PresentRandomUnsafe
```
This should not be in here, even when rare. This can include admin only items.

ArtisticRoomba replied:
Yeah this one's gotta be nuked

slarticodefast replied:
I'm trying to think of the worst items that could be in there.
The kick mines that kick players of the server.
The 9999 TC uplink.
Items that will absolutely kill server performance.
The gib stick is harmless compared to those.

ArtisticRoomba replied:
Tritium/frezon gas miner

I forgot if fixgridatmosmarkers run on spawn (I think they do?) cause if so, fixgridatmosmarkerinstantplasmafire

I actually removed the super synth one when I was refactoring the ent tables but didn't look at anything else in there cause I assumed it was fine.

ScarKy0 replied:
1984
the chances for any of that are near 0, a singular satchel spawns per station and the chance this loot table is even rolled is impossibly low

kosticia commented:
Goddamn
